sas/static/sas/js/picture.js

```diff
@@ -0,0 +1,203 @@
+    /**
+     * The users identified on the currently displayed picture
+     * @type PictureIdentification[]
+     **/
+    identifications: [],
+    /**
+     * The currently displayed picture
+     * Default dummy data are pre-loaded to avoid javascript error
+     * when loading the page at the beginning
+     * @type Picture
+     **/
+    current_picture: {
+      is_moderated: true,
+      id: null,
+      name: "",
+      display_name: "",
+      compressed_url: "",
+      profile_url: "",
+      full_size_url: "",
+      owner: "",
+      date: new Date(),
+    },
+    /**
+     * The picture which will be displayed next if the user press the "next" button
+     * @type ?Picture
+     **/
+    next_picture: null,
+    /**
+     * The picture which will be displayed next if the user press the "previous" button
+     * @type ?Picture
+     **/
+    previous_picture: null,
+    /**
+     * The select2 component used to identify users
+     **/
+    selector: undefined,
+    /**
+     * true if the page is in a loading state, else false
+     **/
+    loading: true,
+    /**
+     * Error message when a moderation operation fails
+     * @type string
+     **/
+    moderation_error: "",
+    /**
+     * Method of pushing new url to the browser history
+     * Used by popstate event and always reset to it's default value when used
+     * @type History
+     **/
+    pushstate: History.PUSH,
+
+    async init() {
+      this.pictures = await fetch_paginated(picture_endpoint);
+      this.selector = sithSelect2({
+        element: $(this.$refs.search),
+        data_source: remote_data_source("/api/user/search", {
+          excluded: () => [...this.identifications.map((i) => i.user.id)],
+          result_converter: (obj) => Object({ ...obj, text: obj.display_name }),
+        }),
+        picture_getter: (user) => user.profile_pict,
+      });
+      this.current_picture = this.pictures.find(
+        (i) => i.id === first_picture_id,
+      );
+      this.$watch("current_picture", () => this.update_picture());
+      window.addEventListener("popstate", async (event) => {
+        if (!event.state || event.state.sas_picture_id === undefined){
+          return;
+        }
+        this.pushstate = History.REPLACE;
+        this.current_picture = this.pictures.find(
+          (i) => i.id === parseInt(event.state.sas_picture_id),
+        );
+      });
+      this.pushstate = History.REPLACE; /* Avoid first url push */
+      await this.update_picture();
+    },
+
+    /**
+     * Update the page.
+     * Called when the `current_picture` property changes.
+     *
+     * The url is modified without reloading the page,
+     * and the previous picture, the next picture and
+     * the list of identified users are updated.
+     */
+    async update_picture() {
+      this.loading = true;
+
+      const update_args = [
+        {sas_picture_id: this.current_picture.id},
+        "",
+        `/sas/picture/${this.current_picture.id}/`,
```

Comment:
Ce serait ptet plus safe de générer ça directement dans le backend et l'envoyer dans le json
On peut avoir un truc genre this.current_picture.sas_url